src/web/client/dal/remoteFetchProvider.ts

```diff
@@ -233,10 +237,27 @@ async function createContentFiles(
             throw new Error(ERROR_CONSTANTS.FILE_NAME_EMPTY);
         }
 
+        let folderName = fileName;
+        if (entityName === schemaEntityName.WEBPAGES) {
+            const webpageNames = WebExtensionContext.getWebpageNames();
+
+            console.log(`Processing webpage: ${fileName}, entityId: ${entityId}, already exists: ${webpageNames.has(fileName)}`);
+
+            if (webpageNames.has(fileName)) {
+                // This is a duplicate - append entity ID
+                folderName = getDuplicateFileName(fileName, entityId);
```

Comment:
Can you add telemetry here? Will be good to see how many times this issue occurs.